src/common/FileSystem.cpp

```diff
@@ -2155,37 +2173,68 @@ bool IsInitialized()
 
-	size_t suffixLen = type == pakType_t::PAK_DIR ? strlen(PAK_DIR_EXT) : strlen(PAK_ZIP_EXT);
 	std::string name, version;
 	Util::optional<uint32_t> checksum;
-	if (!ParsePakName(filename.begin(), filename.end() - suffixLen, name, version, checksum) || (type == pakType_t::PAK_DIR && checksum)) {
-		if (!UseLegacyPaks()) {
-			fsLogs.Warn("Invalid pak name: %s", fullPath);
-			return;
-		} else {
-			fsLogs.Notice("Loading legacy pak: %s", fullPath);
-			name = filename.substr(0, filename.size() - suffixLen);
-			version = "1";
-		}
+
+	if (!ParsePakName(filename.begin(), filename.end(), name, version, checksum) || (isPakDir && checksum)) {
+		fsLogs.Warn("Invalid pak name: '%s'", fullPath);
+		return;
 	}
 
+	if (isPakDir) {
+		fsLogs.Verbose("Found pakdir: '%s'...", fullPath);
+	} else {
+		fsLogs.Verbose("Found pak: '%s'...", fullPath);
+	}
+
+	availablePaks.push_back({std::move(name), std::move(version), checksum, type, std::move(fullPath)});
+}
+
+// Add a legacy pak to the list of available paks
+static void AddLegacyPak(pakType_t type, Str::StringRef filename, Str::StringRef basePath)
+{
+	size_t suffixLen;
+	std::string fullPath = Path::Build(basePath, filename);
+
+	suffixLen = (type == pakType_t::PAK_DIR) ? strlen(LEGACY_PAK_DIR_EXT) : strlen(LEGACY_PAK_ZIP_EXT);
+
+	std::string name, version;
+	Util::optional<uint32_t> checksum;
+
+	name = filename.substr(0, filename.size() - suffixLen);
+	// prefer versioned dpk over legacy pk3 if pak name collides, uses the smallest version string available
+	// this version can't be found in standard dpk, so we can also uses this value to test later if it's a legacy pk3
+	version = "";
+
+	if (type == pakType_t::PAK_ZIP) {
+		fsLogs.Verbose("Found legacy pak: '%s'...", fullPath);
+	} else if (type == pakType_t::PAK_DIR) {
+		fsLogs.Verbose("Found legacy pakdir: '%s'...", fullPath);
+	}
 	availablePaks.push_back({std::move(name), std::move(version), checksum, type, std::move(fullPath)});
 }
 
 // Find all paks in the given path
 static void FindPaksInPath(Str::StringRef basePath, Str::StringRef subPath)
 {
 	std::string fullPath = Path::Build(basePath, subPath);
+	bool useLegacyPaks = UseLegacyPaks();
 	try {
 		for (auto& filename: RawPath::ListFiles(fullPath)) {
 			if (Str::IsSuffix(PAK_ZIP_EXT, filename)) {
 				AddPak(pakType_t::PAK_ZIP, Path::Build(subPath, filename), basePath);
 			} else if (Str::IsSuffix(PAK_DIR_EXT, filename)) {
 				AddPak(pakType_t::PAK_DIR, Path::Build(subPath, filename), basePath);
-			} else if (Str::IsSuffix("/", filename))
+			} else if (useLegacyPaks && Str::IsSuffix(LEGACY_PAK_ZIP_EXT, filename)) {
+				AddLegacyPak(pakType_t::PAK_ZIP, Path::Build(subPath, filename), basePath);
+			} else if (useLegacyPaks && Str::IsSuffix(LEGACY_PAK_DIR_EXT, filename)) {
+				AddLegacyPak(pakType_t::PAK_DIR, Path::Build(subPath, filename), basePath);
+			} else if (Str::IsSuffix("/", filename)) {
 				FindPaksInPath(basePath, Path::Build(subPath, filename));
```

Comment:
Here: https://github.com/DaemonEngine/Daemon/blob/master/src/common/Assert.h#L156 can you add a 
```
#define UNREACHABLE DAEMON_ASSERT_CALLSITE(false, , false, "Unreachable code hit.")
```
and add an else clause containing UNREACHABLE();